sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -1009,12 +1010,12 @@ static ssize_t _tl_authenticate(gcoap_socket_t *sock, const sock_udp_ep_t *remot
     msg_t msg;
     bool is_timed_out = false;
     do {
-        uint32_t start = xtimer_now_usec();
-        res = xtimer_msg_receive_timeout(&msg, timeout);
+        uint32_t start = ztimer_now(ZTIMER_MSEC);
+        res = ztimer_msg_receive_timeout(ZTIMER_MSEC, &msg, timeout);
```

Comment:
The units where changed here, are you sure msec are enough?

Reply:
in all cases where i swap ed timeout of unamend unit (which would be usec in all cases) or usec i checked the typical default values if they mad to msec and only swaped if they did.
There is one exception  GCOAP_OBS_TICK_EXPONENT that can not map to integer msec -> it is still usec